### DISCUSSION TITLE
Cell count from each sample in integrated analysis

### DISCUSSION THREAD

**Discussion by singcell** [satijalab/seurat#5776] [General]
I wanted to determine if the number of particular cell type in a cluster is increasing or decreasing with the treatment. I am using stimulated vs. control PBMC datasets vignette for this analysis.

Question:
I use code below to get the counts in each cluster and treatment condition BUT it shows cell numbers/count from all samples together. I want to know the cell counts in individual samples OR contribution of number of cells from each samples.

`table(Idents(immune.combined), immune.combined$stim)`

Thank you

**Comment by timoast** [comment]
If I'm understanding correctly, you have three different variables: stimulation, batch, and cluster. You want to find the number of cells from each stimulation condition and batch in each cluster? If so, you can add more than two variables in the `table()` function. For example:

```r
counts <- table(Idents(obj), obj$batch, obj$stim)
as.data.frame(counts)
```

this will give you the cell count for each three-way combination (can be scaled to _n_ variables)
